src/Sulu/Bundle/AdminBundle/Admin/NavigationRegistry.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Admin;
+
+use Sulu\Bundle\AdminBundle\Navigation\Navigation;
+use Sulu\Bundle\AdminBundle\Navigation\NavigationItem;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class NavigationRegistry
+{
+    /**
+     * @var Navigation
+     */
+    private $navigation;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var AdminPool
+     */
+    private $adminPool;
+
+    /**
+     * @var RouteRegistry
+     */
+    private $routeRegistry;
+
+    public function __construct(TranslatorInterface $translator, AdminPool $adminPool, RouteRegistry $routeRegistry)
+    {
+        $this->translator = $translator;
+        $this->adminPool = $adminPool;
+        $this->routeRegistry = $routeRegistry;
+    }
+
+    /**
+     * Returns the navigation combined from all Admin objects.
+     */
+    public function getNavigation(): Navigation
+    {
+        if (!$this->navigation) {
+            $this->loadNavigation();
+        }
+
+        return $this->navigation;
+    }
+
+    private function loadNavigation(): void
+    {
+        /** @var Navigation $navigation */
+        $navigation = null;
```

Comment:
If we write `$navigation = new Navigation($rootItem)` you could get rid of this if which is only executed once in the foreach.

Reply:
You have removed my static function..
So I'm not able to get the `navigationRootItem` here..
Therefore it isn't possible here..